x-pack/plugins/search_synonyms/server/plugin.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+  DEFAULT_APP_CATEGORIES,
+} from '@kbn/core/server';
+
+import { KibanaFeatureScope } from '@kbn/features-plugin/common';
+import {
+  SearchSynonymsPluginSetup,
+  SearchSynonymsPluginSetupDependencies,
+  SearchSynonymsPluginStart,
+} from './types';
+
+import { defineRoutes } from './routes';
+import { PLUGIN_ID, PLUGIN_NAME } from '../common';
+
+export class SearchSynonymsPlugin
+  implements Plugin<SearchSynonymsPluginSetup, SearchSynonymsPluginStart, {}, {}>
+{
+  private readonly logger: Logger;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+  }
+
+  public setup(core: CoreSetup, plugins: SearchSynonymsPluginSetupDependencies) {
+    this.logger.debug('searchSynonyms: Setup');
+    const router = core.http.createRouter();
+
+    defineRoutes({ router, logger: this.logger });
+
+    plugins.features.registerKibanaFeature({
+      id: PLUGIN_ID,
+      minimumLicense: 'enterprise',
+      name: PLUGIN_NAME,
+      order: 0,
+      category: DEFAULT_APP_CATEGORIES.enterpriseSearch,
+      app: ['kibana', PLUGIN_ID],
+      scope: [KibanaFeatureScope.Spaces, KibanaFeatureScope.Security],
+      catalogue: [PLUGIN_ID],
+      privileges: {
+        all: {
+          app: ['kibana', PLUGIN_ID],
+          api: [],
+          catalogue: [PLUGIN_ID],
+          savedObject: {
+            all: [],
+            read: [],
+          },
+          ui: [],
+        },
+        read: {
```

Comment:
I experimented a bit on this, I will update these with the next PR where I actually use them otherwise it will complicate. Also will include a cluster permission registration as well since we need those for the APIs.

Do you think changing these freely until we release the feature would cause a problem ? @TattdCodeMonkey